test/e2e/mv3/service-worker-restart.spec.js

```diff
@@ -0,0 +1,188 @@
+
+        // open the account menu
+        await driver.clickElement('[data-testid="account-menu-icon"]');
+        await driver.clickElement({ text: 'Create account', tag: 'div' });
+
+        await driver.fill('.new-account-create-form__input', 'Test Account');
+
+        await driver.clickElement({ text: 'Create', tag: 'button' });
+
+        await driver.openNewPage('chrome://inspect/#service-workers/');
+        await driver.clickElement({
+          text: 'Service workers',
+          tag: 'button',
+        });
+
+        await driver.clickElement({
+          text: 'terminate',
+          tag: 'span',
+        });
+
+        windowHandles = await driver.getAllWindowHandles();
+        const extension = windowHandles[0];
+        await driver.switchToWindow(extension);
+
+        // balance renders
+        await driver.waitForSelector(
+          {
+            css: '[class="eth-overview__primary-container"]',
+            // balance is 0 because we switched to an empty account
+            text: '0 ETH',
+          },
+          { timeout: 50_000 },
+        );
+
+        await driver.findElement({ text: '0x097...7950', tag: 'div' });
+
+        // assert that the segment request has been sent through inspecting the mock
+        await driver.wait(async () => {
+          const isPending = await mockedEndpoint.isPending();
+          return isPending === false;
+        }, 10_000);
+        const mockedRequests = await mockedEndpoint.getSeenRequests();
+
+        assert.equal(mockedRequests.length, numberOfSegmentRequests);
+
+        await assertSWRestartTimeEvent(mockedRequests[0]);
+        await assertSWRestartTimeEvent(mockedRequests[1]);
+        await assertSWProcessActionQueueEvent(
+          mockedRequests[2],
+          'addNewAccount',
+        );
+      },
+    );
+  });
+});
+
+async function assertSWRestartTimeEvent(request) {
+  assert.equal(request.url, 'https://api.segment.io/v1/batch');
+
+  assert.equal(request.body.json.batch.length, 1);
+  assert.equal(
+    request.body.json.batch[0].event,
+    MetaMetricsEventName.ServiceWorkerRestarted,
+  );
+
+  assert.equal(
+    typeof request.body.json.batch[0].properties.service_worker_restarted_time,
+    'number',
+  );
+
+  assert.equal(
+    request.body.json.batch[0].properties.service_worker_restarted_time > 0,
+    true,
+  );
+  assert.equal(
+    request.body.json.batch[0].properties.category,
+    MetaMetricsEventCategory.ServiceWorkers,
+  );
+  assert.equal(
+    request.body.json.batch[0].properties.chain_id,
+    convertToHexValue(1337),
+  );
+  assert.equal(
+    request.body.json.batch[0].properties.environment_type,
+    'background',
+  );
+  assert.equal(request.body.json.batch[0].properties.locale, 'en');
+}
+
+async function assertSWProcessActionQueueEvent(request, method) {
+  assert.equal(request.url, 'https://api.segment.io/v1/batch');
+
+  assert.equal(request.body.json.batch.length, 1);
+
+  assert.equal(
+    request.body.json.batch[0].event,
```

Comment:
Total nit, not a blocker, but you could do `const [firstResult]  = request.body.json.batch;` so you can avoid the long `request.body.json.batch[0]` everywhere.